Sources/Workspace/Workspace+BinaryArtifacts.swift

```diff
@@ -332,24 +332,26 @@ extension Workspace {
                     case .success:
                         observabilityScope.trap { () -> Void in
                             var artifactPath: AbsolutePath? = nil
-                            // strip first level component if needed
-                            if try self.fileSystem.shouldStripFirstLevel(archiveDirectory: tempExtractionDirectory, acceptableExtensions: BinaryTarget.Kind.allCases.map({ $0.fileExtension })) {
-                                observabilityScope.emit(debug: "stripping first level component from  \(tempExtractionDirectory)")
-                                try self.fileSystem.stripFirstLevel(of: tempExtractionDirectory)
-                            } else {
-                                observabilityScope.emit(debug: "no first level component stripping needed for \(tempExtractionDirectory)")
-                            }
-                            let content = try self.fileSystem.getDirectoryContents(tempExtractionDirectory)
-                            // copy from temp location to actual location
-                            for file in content {
-                                let source = tempExtractionDirectory.appending(component: file)
-                                let destination = destinationDirectory.appending(component: file)
-                                if self.fileSystem.exists(destination) {
-                                    try self.fileSystem.removeFileTree(destination)
+                            try self.fileSystem.withLock(on: destinationDirectory, type: .exclusive) {
```

Comment:
this is the key change. we already do this for remote / downloaded archives, this brings it back to local archives as well